util/util.go

```diff
@@ -5,13 +5,44 @@ import (
 	"fmt"
 
 	ipfslite "github.com/hsanjuan/ipfs-lite"
+	"github.com/ipfs/go-datastore"
 	"github.com/ipfs/go-log"
 	crypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
 	"github.com/multiformats/go-multiaddr"
 )
 
-// NewPeer creates a new ipfslite.Peer
-func NewPeer(ctx context.Context, datastorePath string, port int, debug bool) (*ipfslite.Peer, error) {
+// StandardManager is our main implementation of PeerManager
+type StandardManager struct {
+	datastore datastore.Batching
+	host      host.Host
+	dht       *dht.IpfsDHT
+	peer      *ipfslite.Peer
+}
+
+// Peer returns the manager Peer
+func (m *StandardManager) Peer() *ipfslite.Peer {
+	return m.peer
+}
+
+// Stop stops the datastore, libp2p host and dht when requested
+func (m *StandardManager) Stop() error {
+	var lastError error
+	if err := m.datastore.Close(); err != nil {
+		lastError = err
+	}
+	if err := m.host.Close(); err != nil {
+		lastError = err
+	}
+	if err := m.dht.Close(); err != nil {
+		lastError = err
+	}
+	return lastError
+}
+
+// NewPeerManager creates a new server.PeerManager
+func NewPeerManager(ctx context.Context, datastorePath string, port int, debug bool) (*StandardManager, error) {
```

Comment:
I think it makes sense to return a pointer here, but maybe it doesn't actually matter.